service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -656,6 +656,8 @@ func (t *transferQueueActiveTaskExecutor) processSignalExecution(
 			failedCause = enumspb.SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_FAILED_CAUSE_EXTERNAL_WORKFLOW_EXECUTION_NOT_FOUND
 		case *serviceerror.NamespaceNotFound:
 			failedCause = enumspb.SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_FAILED_CAUSE_NAMESPACE_NOT_FOUND
+		case *serviceerror.InvalidArgument:
+			failedCause = enumspb.SIGNAL_EXTERNAL_WORKFLOW_EXECUTION_FAILED_CAUSE_SIGNAL_COUNT_LIMIT_EXCEEDED
```

Comment:
shall we define a specific error type for this? (it is ok to use internal error type)

Reply:
yeah I think that makes sense. Theoretically, that api can return invalid argument error for other reasons like blob size limit exceeded. Right now should be fine though, because the fact that we never handle that error type means other cases never happen and it must be signal count limit. 

I will create a task for adding the new error type, I think it's also useful for other per workflow limit we plan to enforce in the future. For this patch, I think checking invalid argument should be ok.

>  internal error type
I don't quite follow. Do you mean history service returns the newly defined internal error type, and frontend convert it back to invalid argument before returning to customer? Otherwise I don't see how internal error type can work.